invokeai/app/api/routers/models.py

```diff
@@ -54,108 +74,61 @@ async def list_models() -> ModelsList:
-        #         self.generate.model_manager.commit(opt.conf)
-        #         updated_model_list = self.generate.model_manager.list_models()
-        #         socketio.emit(
-        #             "modelDeleted",
-        #             {
-        #                 "deleted_model_name": model_name,
-        #                 "model_list": updated_model_list,
-        #             },
-        #         )
-        #         print(f">> Model Deleted: {model_name}")
-        #     except Exception as e:
-        #         self.handle_exceptions(e)
 
-        # @socketio.on("requestModelChange")
-        # def handle_set_model(model_name: str):
-        #     try:
-        #         print(f">> Model change requested: {model_name}")
-        #         model = self.generate.set_model(model_name)
-        #         model_list = self.generate.model_manager.list_models()
-        #         if model is None:
-        #             socketio.emit(
-        #                 "modelChangeFailed",
-        #                 {"model_name": model_name, "model_list": model_list},
-        #             )
-        #         else:
-        #             socketio.emit(
-        #                 "modelChanged",
-        #                 {"model_name": model_name, "model_list": model_list},
-        #             )
-        #     except Exception as e:
-        #         self.handle_exceptions(e)
+@models_router.post(
+    "/",
+    operation_id="update_model",
+    responses={200: {"status": "success"}},
+)
+async def update_model(
+    model_request: CreateModelRequest
+) -> CreateModelResponse:
+    """ Add Model """
+    model_request_info = model_request.info
+    info_dict = model_request_info.dict()
+    model_response = CreateModelResponse(name=model_request.name, info=model_request.info, status="success")
+
+    ApiDependencies.invoker.services.model_manager.add_model(
+        model_name=model_request.name,
+        model_attributes=info_dict,
+        clobber=True,
+    )
+
+    return model_response
+
+
+@models_router.delete(
+    "/{model_name}",
+    operation_id="del_model",
+    responses={
+        204: {
+        "description": "Model deleted successfully"
+        }, 
+        404: {
+        "description": "Model not found"
+        }
+    },
+)
+async def delete_model(model_name: str) -> None:
+    """Delete Model"""
+    model_names = ApiDependencies.invoker.services.model_manager.model_names()
+    model_exists = model_name in model_names
+
+    # check if model exists
+    print(f">> Checking for model {model_name}...")
+           
+    if model_exists:
+        print(f">> Deleting Model: {model_name}")
+        ApiDependencies.invoker.services.model_manager.del_model(model_name, delete_files=True)
+        print(f">> Model Deleted: {model_name}")
+        raise HTTPException(status_code=204, detail=f"Model '{model_name}' deleted successfully")
+    
+    if not model_exists:
```

Comment:
this should be an `else` instead of `not not model_exists`.

Reply:
I do not not understand. 👍